### PR TITLE
Update usage-rules.md with MStest Analyzers

### PR DESCRIPTION
**Pull request by engyebrahim**
---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/testing/mstest-analyzers/mstest0008.md](https://github.com/dotnet/docs/blob/5c48cf1eca282f94bc79879c033e41b9744fe49f/docs/core/testing/mstest-analyzers/mstest0008.md) | [MSTEST0008: TestInitialize method should have valid layout](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0008?branch=pr-en-us-39592) |
| [docs/core/testing/mstest-analyzers/mstest0009.md](https://github.com/dotnet/docs/blob/5c48cf1eca282f94bc79879c033e41b9744fe49f/docs/core/testing/mstest-analyzers/mstest0009.md) | ["MSTEST0009: TestCleanup method should have valid layout"](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0009?branch=pr-en-us-39592) |
| [docs/core/testing/mstest-analyzers/mstest0010.md](https://github.com/dotnet/docs/blob/5c48cf1eca282f94bc79879c033e41b9744fe49f/docs/core/testing/mstest-analyzers/mstest0010.md) | [MSTEST0010: ClassInitialize method should have valid layout](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0010?branch=pr-en-us-39592) |
| [docs/core/testing/mstest-analyzers/mstest0011.md](https://github.com/dotnet/docs/blob/5c48cf1eca282f94bc79879c033e41b9744fe49f/docs/core/testing/mstest-analyzers/mstest0011.md) | [docs/core/testing/mstest-analyzers/mstest0011](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0011?branch=pr-en-us-39592) |
| [docs/core/testing/mstest-analyzers/mstest0012.md](https://github.com/dotnet/docs/blob/5c48cf1eca282f94bc79879c033e41b9744fe49f/docs/core/testing/mstest-analyzers/mstest0012.md) | [MSTEST0012: AssemblyInitialize method should have valid layout](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0012?branch=pr-en-us-39592) |
| [docs/core/testing/mstest-analyzers/mstest0013.md](https://github.com/dotnet/docs/blob/5c48cf1eca282f94bc79879c033e41b9744fe49f/docs/core/testing/mstest-analyzers/mstest0013.md) | ["MSTEST0013: AssemblyCleanup method should have valid layout"](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0013?branch=pr-en-us-39592) |
| [docs/core/testing/mstest-analyzers/usage-rules.md](https://github.com/dotnet/docs/blob/5c48cf1eca282f94bc79879c033e41b9744fe49f/docs/core/testing/mstest-analyzers/usage-rules.md) | [MSTest usage rules](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/usage-rules?branch=pr-en-us-39592) |

### CODE REVIEW

File: docs/core/testing/mstest-analyzers/mstest0008.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- it should not be `async void`
+- it should not be `static`
+- it should not be a special method (finalizer, operator...).
+- it should not be generic
+- it should not take any parameter
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0008.md:45]:
```suggestion
Ensure that the method matches the layout described above.
```

File: docs/core/testing/mstest-analyzers/mstest0009.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+title: "MSTEST0009: `TestCleanup` should be valid"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0009.md:2]:
```suggestion
title: "MSTEST0009: TestCleanup method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0009.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+description: "Learn about code analysis rule MSTEST0009: Ensure `TestCleanup` has a valid layout"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0009.md:3]:
```suggestion
description: "Learn about code analysis rule MSTEST0009: TestCleanup method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0009.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- MSTEST0009
+author: engyebrahim
+ms.author: enjieid
+---
+# MSTEST0009: `TestCleanup` should be valid
+
+| Property                            | Value                                    |
+|-------------------------------------|------------------------------------------|
+| **Rule ID**                         | MSTEST0009                               |
+| **Title**                           | `TestCleanup` should be valid         |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0009.md:19]:
```suggestion
| **Title**                           | TestCleanup method should have valid layout         |
```

File: docs/core/testing/mstest-analyzers/mstest0009.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+- it should not take any parameter
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
+
+## When to suppress warnings
+
+Do not suppress a warning from this rule. If you ignore this rule, your tests woun't run.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0009.md:49]:
```suggestion
Do not suppress a warning from this rule. If you ignore this rule, flagged instances will be either skipped or result in runtime error.
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+- it should take one parameter of type `TestContext`
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
+
+## When to suppress warnings
+
+Do not suppress a warning from this rule. If you ignore this rule, your tests woun't run.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:48]:
```suggestion
Do not suppress a warning from this rule. If you ignore this rule, flagged instances will be either skipped or result in runtime error.
```

File: docs/core/testing/mstest-analyzers/mstest0012.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+- it should take one parameter of type `TestContext`
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
+
+## When to suppress warnings
+
+Do not suppress a warning from this rule. If you ignore this rule, your tests woun't run.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0012.md:48]:
```suggestion
Do not suppress a warning from this rule. If you ignore this rule, flagged instances will be either skipped or result in runtime error.
```

File: docs/core/testing/mstest-analyzers/mstest0009.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+- it should not be `async void`
+- it should not be `static`
+- it should not be a special method (finalizer, operator...).
+- it should not be generic
+- it should not take any parameter
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0009.md:45]:
```suggestion
Ensure that the method matches the layout described above.
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+- it should be `static`
+- it should not be `async void`
+- it should not be a special method (finalizer, operator...).
+- it should not be generic
+- it should take one parameter of type `TestContext`
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:44]:
```suggestion
Ensure that the method matches the layout described above.
```

File: docs/core/testing/mstest-analyzers/mstest0011.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+- it should be `static`
+- it should not be `async void`
+- it should not be a special method (finalizer, operator...).
+- it should not be generic
+- it should not take any parameter
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0011.md:44]:
```suggestion
Ensure that the method matches the layout described above.
```

File: docs/core/testing/mstest-analyzers/mstest0012.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+- it should be `static`
+- it should not be `async void`
+- it should not be a special method (finalizer, operator...).
+- it should not be generic
+- it should take one parameter of type `TestContext`
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0012.md:44]:
```suggestion
Ensure that the method matches the layout described above.
```

File: docs/core/testing/mstest-analyzers/mstest0013.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+- it should be `static`
+- it should not be `async void`
+- it should not be a special method (finalizer, operator...).
+- it should not be generic
+- it should not take any parameter
+- return type should be `void`, `Task` or `ValueTask`
+
+## How to fix violations
+
+To fix a violation of this rule, you should fix the layout.
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0013.md:44]:
```suggestion
Ensure that the method matches the layout described above.
```

File: docs/core/testing/mstest-analyzers/mstest0013.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+- MSTEST0009
+author: engyebrahim
+ms.author: enjieid
+---
+# MSTEST0009: `AssemblyCleanup` should be valid
+
+| Property                            | Value                                    |
+|-------------------------------------|------------------------------------------|
+| **Rule ID**                         | MSTEST0009                               |
+| **Title**                           | `AssemblyCleanup` should be valid         |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0013.md:19]:
```suggestion
| **Title**                           | AssemblyCleanup method should have valid layout         |
```

File: docs/core/testing/mstest-analyzers/mstest0013.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `AssemblyCleanup` should be valid"
+description: "Learn about code analysis rule MSTEST0009: Ensure `AssemblyCleanup` has a valid layout"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0013.md:3]:
```suggestion
description: "Learn about code analysis rule MSTEST0013: AssemblyCleanup method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0013.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `AssemblyCleanup` should be valid"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0013.md:2]:
```suggestion
title: "MSTEST0013: AssemblyCleanup method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0012.md [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+- MSTEST0009
+author: engyebrahim
+ms.author: enjieid
+---
+# MSTEST0009: `AssemblyInitialize` should be valid
+
+| Property                            | Value                                    |
+|-------------------------------------|------------------------------------------|
+| **Rule ID**                         | MSTEST0009                               |
+| **Title**                           | `AssemblyInitialize` should be valid         |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0012.md:19]:
```suggestion
| **Title**                           | AssemblyInitialize method should have valid layout        |
```

File: docs/core/testing/mstest-analyzers/mstest0013.md [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+description: "Learn about code analysis rule MSTEST0009: Ensure `AssemblyCleanup` has a valid layout"
+ms.date: 02/19/2024
+f1_keywords:
+- MSTEST0009
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0013.md:6]:
```suggestion
- MSTEST00013
```

File: docs/core/testing/mstest-analyzers/mstest0013.md [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+- AssemblyCleanupShouldBeValidAnalyzer
+helpviewer_keywords:
+- AssemblyCleanupShouldBeValidAnalyzer
+- MSTEST0009
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0013.md:10]:
```suggestion
- MSTEST0013
```

File: docs/core/testing/mstest-analyzers/mstest0013.md [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+author: engyebrahim
+ms.author: enjieid
+---
+# MSTEST0009: `AssemblyCleanup` should be valid
+
+| Property                            | Value                                    |
+|-------------------------------------|------------------------------------------|
+| **Rule ID**                         | MSTEST0009                               |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0013.md:18]:
```suggestion
| **Rule ID**                         | MSTEST0013                               |
```

File: docs/core/testing/mstest-analyzers/mstest0012.md [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+- AssemblyInitializeShouldBeValidAnalyzer
+- MSTEST0009
+author: engyebrahim
+ms.author: enjieid
+---
+# MSTEST0009: `AssemblyInitialize` should be valid
+
+| Property                            | Value                                    |
+|-------------------------------------|------------------------------------------|
+| **Rule ID**                         | MSTEST0009                               |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0012.md:18]:
```suggestion
| **Rule ID**                         | MSTEST0012                               |
```

File: docs/core/testing/mstest-analyzers/mstest0012.md [hunk 21]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `AssemblyInitialize` should be valid"
+description: "Learn about code analysis rule MSTEST0009: Ensure `AssemblyInitialize` has a valid layout"
+ms.date: 02/19/2024
+f1_keywords:
+- MSTEST0009
+- AssemblyInitializeShouldBeValidAnalyzer
+helpviewer_keywords:
+- AssemblyInitializeShouldBeValidAnalyzer
+- MSTEST0009
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0012.md:10]:
```suggestion
- MSTEST0012
```

File: docs/core/testing/mstest-analyzers/mstest0012.md [hunk 22]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `AssemblyInitialize` should be valid"
+description: "Learn about code analysis rule MSTEST0009: Ensure `AssemblyInitialize` has a valid layout"
+ms.date: 02/19/2024
+f1_keywords:
+- MSTEST0009
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0012.md:6]:
```suggestion
- MSTEST0012
```

File: docs/core/testing/mstest-analyzers/mstest0012.md [hunk 23]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `AssemblyInitialize` should be valid"
+description: "Learn about code analysis rule MSTEST0009: Ensure `AssemblyInitialize` has a valid layout"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0012.md:3]:
```suggestion
description: "Learn about code analysis rule MSTEST0012: AssemblyInitialize method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0012.md [hunk 24]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `AssemblyInitialize` should be valid"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0012.md:2]:
```suggestion
title: "MSTEST0012: AssemblyInitialize method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0011.md [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+- ClassCleanupShouldBeValidAnalyzer
+- MSTEST0009
+author: engyebrahim
+ms.author: enjieid
+---
+# MSTEST0009: `ClassCleanup` should be valid
+
+| Property                            | Value                                    |
+|-------------------------------------|------------------------------------------|
+| **Rule ID**                         | MSTEST0009                               |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0011.md:18]:
```suggestion
| **Rule ID**                         | MSTEST0011                               |
```

File: docs/core/testing/mstest-analyzers/mstest0011.md [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+| **Title**                           | `ClassCleanup` should be valid         |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0011.md:19]:
```suggestion
| **Title**                           | ClassCleanup method should have valid layout         |
```

File: docs/core/testing/mstest-analyzers/mstest0011.md [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+f1_keywords:
+- MSTEST0009
+- ClassCleanupShouldBeValidAnalyzer
+helpviewer_keywords:
+- ClassCleanupShouldBeValidAnalyzer
+- MSTEST0009
+author: engyebrahim
+ms.author: enjieid
+---
+# MSTEST0009: `ClassCleanup` should be valid
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0011.md:14]:
```suggestion
# MSTEST0011: ClassCleanup method should have valid layout
```

File: docs/core/testing/mstest-analyzers/mstest0011.md [hunk 28]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `ClassCleanup` should be valid"
+description: "Learn about code analysis rule MSTEST0009: Ensure `ClassCleanup` has a valid layout"
+ms.date: 02/19/2024
+f1_keywords:
+- MSTEST0009
+- ClassCleanupShouldBeValidAnalyzer
+helpviewer_keywords:
+- ClassCleanupShouldBeValidAnalyzer
+- MSTEST0009
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0011.md:10]:
```suggestion
- MSTEST0011
```

File: docs/core/testing/mstest-analyzers/mstest0011.md [hunk 29]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `ClassCleanup` should be valid"
+description: "Learn about code analysis rule MSTEST0009: Ensure `ClassCleanup` has a valid layout"
+ms.date: 02/19/2024
+f1_keywords:
+- MSTEST0009
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0011.md:6]:
```suggestion
- MSTEST0011
```

File: docs/core/testing/mstest-analyzers/mstest0011.md [hunk 30]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `ClassCleanup` should be valid"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0011.md:2]:
```suggestion
title: "MSTEST0011: ClassCleanup method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0011.md [hunk 31]
```diff
@@ -0,0 +1,48 @@
...
+description: "Learn about code analysis rule MSTEST0009: Ensure `ClassCleanup` has a valid layout"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0011.md:3]:
```suggestion
description: "Learn about code analysis rule MSTEST0011: ClassCleanup method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 32]
```diff
@@ -0,0 +1,48 @@
+---
+title: "MSTEST0009: `ClassInitialize` should be valid"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:2]:
```suggestion
title: "MSTEST0010: ClassInitialize method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 33]
```diff
@@ -0,0 +1,48 @@
...
+description: "Learn about code analysis rule MSTEST0009: Ensure `ClassInitialize` has a valid layout"
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:3]:
```suggestion
description: "Learn about code analysis rule MSTEST0010: ClassInitialize method should have valid layout"
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 34]
```diff
@@ -0,0 +1,48 @@
...
+ms.date: 02/19/2024
+f1_keywords:
+- MSTEST0009
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:6]:
```suggestion
- MSTEST0010
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 35]
```diff
@@ -0,0 +1,48 @@
...
+- ClassInitializeShouldBeValidAnalyzer
+helpviewer_keywords:
+- ClassInitializeShouldBeValidAnalyzer
+- MSTEST0009
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:10]:
```suggestion
- MSTEST0010
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 36]
```diff
@@ -0,0 +1,48 @@
...
+author: engyebrahim
+ms.author: enjieid
+---
+# MSTEST0009: `ClassInitialize` should be valid
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:14]:
```suggestion
# MSTEST0010: ClassInitialize method should have valid layout
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 37]
```diff
@@ -0,0 +1,48 @@
...
+
+| Property                            | Value                                    |
+|-------------------------------------|------------------------------------------|
+| **Rule ID**                         | MSTEST0009                               |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:18]:
```suggestion
| **Rule ID**                         | MSTEST0010                               |
```

File: docs/core/testing/mstest-analyzers/mstest0010.md [hunk 38]
```diff
@@ -0,0 +1,48 @@
...
+| **Title**                           | `ClassInitialize` should be valid         |
```
Comment (Evangelink) [inline docs/core/testing/mstest-analyzers/mstest0010.md:19]:
```suggestion
| **Title**                           | ClassInitialize method should have valid layout |
```

**Review by Evangelink — COMMENTED** [review]
Please review again all docs, I might have missed some entries because there were many copy-paste issues.

**Review by Evangelink — APPROVED** [review]

**Comment by Evangelink** [comment]
@IEvangelist PR is ready for review when you have time.

Thank you @engyebrahim

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Pull request merged**